tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,157 @@
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed())
+					Eventually(createIstioCR).WithArguments(version).Should(Succeed(), "Istio CR should be created")
+					GinkgoWriter.Println("Istio CR created")
+				})
+
+				AfterAll(func() {
+					GinkgoWriter.Println("Cleaning up")
+					Eventually(kubectl.DeleteNamespace).WithArguments(controlPlaneNamespace).Should(Succeed(), "Namespace should be deleted")
+					Eventually(kubectl.CheckNamespaceExist).WithArguments(controlPlaneNamespace).Should(MatchError(kubectl.ErrNotFound), "Namespace should be deleted")
+					GinkgoWriter.Println("Cleanup done")
+				})
+
+				When("the Istio resource is created", func() {
+					It("updates the Istio resource status to Ready and Running", func() {
+						Eventually(kubectl.GetResourceCondition).WithArguments(controlPlaneNamespace, "istio", istioName).Should(ContainElement(resourceReconcilied))
+						Eventually(kubectl.GetResourceCondition).WithArguments(controlPlaneNamespace, "istio", istioName).Should(ContainElement(resourceReady))
+						Eventually(kubectl.GetPodPhase).WithArguments(controlPlaneNamespace, "app=istiod").Should(Equal("Running"), "Istiod should be Running")
+					})
+
+					Specify("the istio resource version match the applied version", func() {
+						// TODO: we need to add a function to get the istio version from the control panel directly
+						// and compare it with the applied version
+						// This is a TODO because actual version.yaml contains for example latest and not the actual version
+						// Posible solution is to add actual version to the version.yaml
+					})
+
+					Specify("istio resource stopped reconciling", func() {
+						istiodPodName, _ := kubectl.GetPodFromLabel(controlPlaneNamespace, "app=istiod")
+						Eventually(kubectl.GetPodLogs).WithArguments(controlPlaneNamespace, istiodPodName, "30s").ShouldNot(ContainSubstring("Reconciliation done"))
+						GinkgoWriter.Println("Istio Operator stopped reconciling")
+					})
+
+					It("deploys istiod", func() {
+						expectedDeployments := []string{"istiod"}
+
+						deploymentsList, err := kubectl.GetDeployments(controlPlaneNamespace)
+						if err != nil {
+							Fail("Error getting deployments")
+						}
+
+						Expect(deploymentsList).To(Equal(expectedDeployments), "Deployments List should contains only istiod")
+					})
+
+					It("deploys the CNI DaemonSet when running on OpenShift", func() {
+						if ocp == "true" {
+							Eventually(kubectl.GetDaemonSets).WithArguments(namespace).Should(ContainElement("istio-cni-node"), "CNI DaemonSet should be deployed")
+							Eventually(kubectl.GetPodPhase).WithArguments(namespace, "k8s-app=istio-cni-node").Should(Equal("Running"), "CNI DaemonSet should be Running")
+							GinkgoWriter.Println("CNI DaemonSet is deployed in the namespace and Running")
+						} else {
+							Eventually(kubectl.GetDaemonSets).WithArguments(namespace).Should(BeEmpty(), "CNI DaemonSet should not be deployed on OpenShift")
```

Comment:
This should either use `Consistently` or just `Expect`. Using `Eventually(...).Should(BeEmpty())` for something that *shouldn't* happen, is incorrect. Especially in asynchronous systems. For example, this assertion will pass even i f the operator eventually deploys the DaemonSet (e.g. after a few seconds). For things that musn't happen, you should use `Consistently` (but we shouldn't overuse them, since they slow down the tests a lot).